### PR TITLE
Add a set of sensible default settings

### PR DESCRIPTION
**Pull request by paddyroddy**
Have installed this app and picked some sensible defaults https://github.com/repository-settings/app

Looks like we can inherit from this for other repos, need to work out how to do that.

### CODE REVIEW

**Comment by p-j-smith** [comment]
these settings would be super useful! hopefully the app is still supported / functioning

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    color: 9E2383
+  - name: wontfix
+    description: This will not be worked on
+    color: FFFFFF
+
+repository:
+  allow_auto_merge: true
+  allow_merge_commit: false
+  allow_rebase_merge: false
+  allow_update_branch: true
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:108]: image
It's this setting

File: .github/settings.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  default_branch: main
+  delete_branch_on_merge: true
+  has_downloads: false
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:111]: Some deprecated GitHub setting https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository#argument-reference feels like we may as well turn it off

File: .github/settings.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  has_wiki: false
+  squash_merge_commit_message: BLANK
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:113]: Tbh I'm not sure what this setting does. Let me have another look

[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml]: Gone for `PR_BODY`

File: .github/settings.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  - name: ucl-internal
+    description:
+    color: 638AB6
+  - name: ucl-ncita
+    description:
+    color: 5F9460
+  - name: ucl-open
+    description:
+    color: D6BCAA
+  - name: WEISS
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:97]: No idea. These are copied across from https://github.com/UCL-MIRSG/project-records/labels. Let's get this in, they can be refined if it works.

File: .github/settings.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  - name: invalid
+    description: This doesn't seem right
+    color: E4E669
+  - name: ION
+    description:
+    color: 2CD3F9
+  - name: LHA
+    description:
+    color: A85482
+  - name: NCITA
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:64]: No idea. These are copied across from https://github.com/UCL-MIRSG/project-records/labels. Let's get this in, they can be refined if it works.

File: .github/settings.yml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      required_pull_request_reviews:
+        required_approving_review_count: 1
+        bypass_pull_request_allowances:
+          apps:
+            - renovate
+      required_status_checks:
+        strict: true
+        contexts:
+          - linting
+      enforce_admins: true
```
[the remark this answers is not in the document]
Comment (paddyroddy) [reply .github/settings.yml:13]: We'll have to enable the app

**Comment by paddyroddy** [comment]
> will these setting be applied to all existing repos? If so, what will happen if e.g. a repo doesn't have a `linting` workflow (it looks like these settings will require a `linting` status check to pass)?

So this is only for this repo. Looks like we can inherit, which I need to test once this is in.

**Review by p-j-smith — APPROVED** [review]
okay cool, let's try it on this repo. Although I'm a little hesitant about enabling it on all repos through inheriting!

**Comment by paddyroddy** [comment]
> okay cool, let's try it on this repo. Although I'm a little hesitant about enabling it on all repos through inheriting!

Well I haven't done that, so literally nothing to worry. Read more here https://github.com/repository-settings/app#inheritance

**Pull request merged**